### PR TITLE
script.py: ensure all output is written

### PR DESCRIPTION
**Pull request by aerickson**
See https://bugzilla.mozilla.org/show_bug.cgi?id=1595129 for background.

changes:
- ensure the full line has been written
- add additional debugging output to determine where this script might be hanging
- upgrade to mozdevice 3.0.6 for py3 compatibility
- pin docker image to a digest value
  - https://medium.com/@tariq.m.islam/container-deployments-a-lesson-in-deterministic-ops-a4a467b14a03

Test plan:
- Build locally and ensure script.py and entrypoint.sh run.
  - Working.
- Build an image from this branch and do initial Bitbar testing.
  - test links below

Images built from this PR/branch: 
- fce77ad : `Successfully tagged mozilla-image:20191114T161717`, https://mozilla.testdroid.com/#testing/device-session/1328432/1385525/35780048
  - tests: https://treeherder.mozilla.org/#/jobs?repo=try&revision=1635c9df8268cbe29f6551ef7cc2b665c95a4a60
- 471e227 : `Successfully tagged mozilla-image:20191119T130125`, https://mozilla.testdroid.com/#testing/test-run/1328432/1403186
  - tests
    - g5: https://treeherder.mozilla.org/#/jobs?repo=try&revision=4f042d80ceab83b3af76d8a75e13424cc006c596
    - p2: https://treeherder.mozilla.org/#/jobs?repo=try&revision=3b84ad0f30e53f2e4a2b7771981b4ae4a7443f1c

reference try pushes:
- g5: https://treeherder.mozilla.org/#/jobs?repo=try&revision=575f1f6057604350ca583bdea6e92dbaa2ea9b22
- p2: https://treeherder.mozilla.org/#/jobs?repo=try&revision=bfe03f683fc6b8e38d5829b755f6172561de2206

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -120,11 +120,18 @@ RUN cd /tmp && \
     tar xzf /builds/worker/Downloads/android-sdk_r24.3.4-linux.tgz --directory=/builds/worker || true && \
     unzip -qq -n /builds/worker/Downloads/sdk-tools-linux-4333796.zip -d /builds/worker/android-sdk-linux/ || true && \
     /builds/worker/android-sdk-linux/tools/bin/sdkmanager platform-tools "build-tools;28.0.3" && \
+    # upgrade the builtin setuptools
     pip install setuptools -U && \
-    pip install mozdevice==3.0.5 && \
+    pip3 install setuptools -U && \
+    # pips used by scripts in this docker image
```
Comment (bclary) [inline Dockerfile:126]: you need a trailing escape after the comment in order to keep the pips all in the overall command line.

File: taskcluster/script.py [hunk 2]
```diff
@@ -53,6 +53,7 @@ def get_device_type(device):
 
 
 def enable_charging(device, device_type):
+    print("script.py: enabling charging...")
```
Comment (bclary) [inline taskcluster/script.py]: Might as well print the device and device_type.

**Review by bclary — CHANGES_REQUESTED** [review]
minor changes.

File: Dockerfile (hunk 1 above)
Comment (aerickson) [reply Dockerfile:126]: The Dockerfile builds locally.

https://docs.docker.com/engine/reference/builder/#format indicates that "Line continuation characters are not supported in comments."

File: taskcluster/script.py (hunk 2 above)
Comment (aerickson) [reply taskcluster/script.py]: Done.

File: Dockerfile (hunk 1 above)
Comment (aerickson) [reply Dockerfile:126]: The comments seem to be dropped. Command run when building:

Step 23/25 : RUN cd /tmp &&     chmod +x /usr/local/bin/generic-worker &&     chmod +x /usr/local/bin/livelog &&     chmod +x /usr/local/bin/tooltool.py &&     chmod +x /usr/local/bin/entrypoint.* &&     chmod +x /builds/taskcluster/script.py &&     mkdir /root/.android &&     touch /root/.android/repositories.cfg &&     tar xzf /builds/worker/Downloads/node-v8.11.3-linux-x64.tar.gz -C /usr/local --strip-components 1 &&     node -v &&     npm -v &&     tar xzf /builds/worker/Downloads/android-sdk_r24.3.4-linux.tgz --directory=/builds/worker || true &&     unzip -qq -n /builds/worker/Downloads/sdk-tools-linux-4333796.zip -d /builds/worker/android-sdk-linux/ || true &&     /builds/worker/android-sdk-linux/tools/bin/sdkmanager platform-tools "build-tools;28.0.3" &&     pip install setuptools -U &&     pip3 install setuptools -U &&     pip install google-cloud-logging &&     pip3 install google-cloud-logging &&     pip install mozdevice==3.0.5 &&     pip3 install mozdevice==3.0.5 &&     pip install zstandard==0.11.1 &&     pip3 install zstandard==0.11.1 &&     rm -rf /tmp/* &&     rm -rf /var/lib/apt/lists/* &&     rm -rf /builds/worker/Downloads/* &&     chown -R root:worker /builds &&     chmod 775 /builds

**Review by bclary — APPROVED** [review]
r+. Thanks! Dockerfiles are weird.

**Comment by aerickson** [comment]
Sorry, found a bug. 

Changes since your approval: https://github.com/aerickson/mozilla-bitbar-docker/compare/6b392d259678c11966b44ce3685b882fe68c9652...aerickson:fce77ad6edc39412fbe9c72cbc0eddcd2d78ec17

**Comment by aerickson** [comment]
I had commented out much of the adb device interaction code for local testing. 

It seems mozdevice isn't compatible with python 3.

from the test run:

```
[task 2019-11-15T19:41:37.446Z] Traceback (most recent call last):
[task 2019-11-15T19:41:37.447Z]   File "/builds/taskcluster/script.py", line 245, in <module>
[task 2019-11-15T19:41:37.447Z]     sys.exit(main())
[task 2019-11-15T19:41:37.447Z]   File "/builds/taskcluster/script.py", line 157, in main
[task 2019-11-15T19:41:37.447Z]     devices = adbhost.devices()
[task 2019-11-15T19:41:37.447Z]   File "/usr/local/lib/python3.6/dist-packages/mozdevice/adb.py", line 509, in devices
[task 2019-11-15T19:41:37.447Z]     match = re_device_info.match(line)
[task 2019-11-15T19:41:37.447Z] TypeError: cannot use a string pattern on a bytes-like object
```

**Comment by aerickson** [comment]
Updated diff since last approval: https://github.com/aerickson/mozilla-bitbar-docker/compare/6b392d259678c11966b44ce3685b882fe68c9652...aerickson:ac5f1d6

**Review by bclary — APPROVED** [review]
r+ seems fine to me.

**Comment by aerickson** [comment]
Results from comparing the baseline runs to the new runs:

All times for the new image match the reference image (outliers were rerun and now have matching times), except for one g5 job (https://treeherder.mozilla.org/#/jobs?repo=try&selectedJob=277060970&group_state=expanded&revision=4f042d80ceab83b3af76d8a75e13424cc006c596) that went from 13 minutes to 30 minutes. I'm rerunning it.

**Comment by aerickson** [comment]
> Results from comparing the baseline runs to the new runs:
> 
> All times for the new image match the reference image (outliers were rerun and now have matching times), except for one g5 job (https://treeherder.mozilla.org/#/jobs?repo=try&selectedJob=277060970&group_state=expanded&revision=4f042d80ceab83b3af76d8a75e13424cc006c596) that went from 13 minutes to 30 minutes. I'm rerunning it.

The reruns have finished and that job is back to the base timing.

**Pull request merged**